### PR TITLE
Compose: Use entities in priority and strategy selectors, refactor

### PR DESCRIPTION
**Pull request by mikapfl**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Description in a ``.rst`` file in the directory ``changelog_unreleased`` added - remember to include a ``*`` to make it a bullet point

## Description

* general cleaning up of `compose`
* strategy and priority selectors:
    - support and test multiple values per dimension (priority: not for priority dimensions)
    - support selecting by entity (not for priority dimensions)

### CODE REVIEW

**Comment by mikapfl** [comment]
Should be merged after #199 , also the changes view in github doesn't make a lot of sense currently because it compares to the main branch, will make more sense when #199 is merged.

File: primap2/csg/_compose.py [hunk 1]
```diff
@@ -77,52 +78,52 @@ def compose(
...
             pbar = None
         else:
             pbar = progress_bar(total=number_of_timeseries, unit="ts", unit_scale=True)
+
         iterate_next_fixed_dimension(
             input_da=input_da,
-            priority_definition=priority_definition,
-            strategy_definition=strategy_definition,
+            priority_definition=priority_definition.limit("entity", entity),
+            strategy_definition=strategy_definition.limit("entity", entity),
```
Comment (mikapfl) [inline primap2/csg/_compose.py:126]: it was surprisingly easy to enable filtering by entity, these two lines are all that's needed.

Comment (JGuetschow) [reply primap2/csg/_compose.py:130]: Nice. didn't expect that but in hindsight I could have expected it as we're working on a per entity basis anyway, so no need for additional filtering of entities.

File: setup.cfg [hunk 2]
```diff
@@ -77,6 +77,7 @@ dev =
     mypy
     tox
     numpydoc
+    ruff
```
Comment (JGuetschow) [inline setup.cfg:80]: So for ruff is not run by default, right? I expect a lot of formatting changes and complaint about future deprecations when running ruff on the existing code. (We have to do it a some point ...)

**Review by JGuetschow — APPROVED** [review]

File: setup.cfg (hunk 2 above)
Comment (mikapfl) [reply setup.cfg:80]: I'm running ruff on the repo by default, it doesn't do any formatting changes anymore, they have been done already. They were minor (it is pretty close to black's style).

With the default ruleset, also `ruff check` is clean. I haven't checked with the CR ruleset.

Comment (mikapfl) [reply setup.cfg:80]: Maybe this wasn't clear: We are running ruff already in the pre-commit (so also on github in the checks) in both checking and formatting mode. We have a reduced set of ruff check rules active compared to other CR projects. If you added the documentation rules and some other stuff, we would get lots of warnings, some of them actually things we should fix, some of them just stupid. At some point I could look through this, but I don't think it is pressing.

**Pull request merged**